### DISCUSSION TITLE
How to disable @odata.type Field in serialized Object

### DISCUSSION THREAD

**Discussion by TobiasForbrich** [OData/odata.net#3011] [Q&A]
Hi community,

sorry for my bad english and probably stupid question: I am using the new OData Client for .NET 8.0 and have following issue:

I have a given OData Rest API Service with a Object "ProductionOrder" which has several Enum-Properties. 

When I create a POST Message using the DataServiceContext Client-Classes generated by the ConnectedServices Visual Studio Plug-In, the serialized JSON-Body contains extra "@odata.type" fields for each Enum-Member of the ProductionOrder. Here a simplified example with the enum-Property "Status"

`{
    "@odata.type": "#MESOGanttplan.Api.Module.BusinessObjects.ProductionOrder",
    "ProductionOrderId": "test2",
    "Status@odata.type": "#MESOGanttplan.Model.Enums.ProductionOrderStatusEnum",
    "Status": "NotPlanned"
}`

The Backend OData Service I use can't handle this extra properties and fails. 
If I manually remove the "Status@odata.type" property, like following, the request succeeds. 

`{
    "@odata.type": "#MESOGanttplan.Api.Module.BusinessObjects.ProductionOrder",
    "ProductionOrderId": "test2",
    "Status": "NotPlanned"
}`

Is there any way to avoid the extra Json-Field for the enum-Propertytypes?

A am happy for any advice!
regards,
Tobias

**Comment by gathogojr** [comment]
Hello @TobiasForbrich. Is the `Status` enum property a dynamic property by any chance?

**Reply by gathogojr** [reply]
Hello @TobiasForbrich. I tested this and confirmed that OData client includes `@odata.type` property annotations for both declared and dynamic enum properties. This is odd behaviour because we don't do the same for other declared properties.

It'd probably be a breaking change to rectify it in a minor version but we can consider it for the next major release
